pyccel/ast/numpyext.py

```diff
@@ -881,30 +882,59 @@ def fprint(self, printer, lhs):
         return alloc +'\n' + stmt
 
 #==============================================================================
-class Rand(Real):
+class Rand(Function, PyccelAstNode):
 
     """
       Represents a call to  numpy.random.random or numpy.random.rand for code generation.
 
     """
-    def __new__(cls, arg = Nil()):
-        return Real.__new__(cls, arg)
+    _dtype = 'real'
+    _precision = default_precision['real']
 
-    def __init__(self):
-        Real.__init__(self, None)
+    def __init__(self, *args):
+        if len(args) == 0:
+            self._shape = ()
```

Comment:
Isn't `args` an empty tuple when its length is zero?

Reply:
You are right!